randovania/game_connection/executor/am2r_executor.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import asyncio
+import dataclasses
+import logging
+import struct
+import typing
+from asyncio import StreamReader, StreamWriter
+from enum import IntEnum
+
+from PySide6.QtCore import QObject, Signal
+
+
+class AM2RConnectionException(Exception):
+    pass
+
+
+@dataclasses.dataclass()
+class AM2RSocketHolder:
+    reader: StreamReader
+    writer: StreamWriter
+    api_version: int
+    buffer_size: int
+    request_number: int
```

Comment:
This becomes a repeating pattern. If MSR multiworld ever gonna to happen, we would add the same.
Would be nice to refactor it to a common Socketholder thing for dread and am2r now.

Reply:
I'll just add a TODO for now :sweat_smile:

Reply:
😔